lib/ansible/plugins/inventory/vultr.py

```diff
@@ -0,0 +1,151 @@
+    'auto_backups': dict(key='auto_backup_enabled', convert_to='bool'),
+    'current_bandwidth_gb': dict(),
+    'kvm_url': dict(),
+    'default_password': dict(),
+    'internal_ip': dict(),
+    'disk': dict(),
+    'cost_per_month': dict(convert_to='float'),
+    'location': dict(key='region'),
+    'main_ip': dict(key='v4_main_ip'),
+    'network_v4': dict(key='v4_network'),
+    'gateway_v4': dict(key='v4_gateway'),
+    'os': dict(),
+    'pending_charges': dict(convert_to='float'),
+    'power_status': dict(),
+    'ram': dict(),
+    'plan': dict(),
+    'server_state': dict(),
+    'status': dict(),
+    'firewall_group': dict(),
+    'tag': dict(),
+    'v6_main_ip': dict(),
+    'v6_network': dict(),
+    'v6_network_size': dict(),
+    'v6_networks': dict(),
+    'vcpu_count': dict(convert_to='int'),
+}
+
+
+def _load_conf(path, account):
+
+    if path:
+        conf = configparser.ConfigParser()
+        conf.read(path)
+
+        if not conf._sections.get(account):
+            return None
+
+        return dict(conf.items(account))
+    else:
+        return Vultr.read_ini_config(account)
+
+
+def _retrieve_servers(api_key):
+    api_url = 'https://api.vultr.com/v1/server/list'
+
+    try:
+        response = open_url(
+            api_url, headers={'API-Key': api_key, 'Content-type': 'application/json'},
+            http_agent=VULTR_USER_AGENT,
+        )
+        servers_list = json.loads(response.read())
+        if not servers_list:
+            return []
+
+        return [server for id, server in servers_list.items()]
+    except ValueError as e:
+        raise AnsibleError("Incorrect JSON payload")
+    except Exception as e:
+        raise AnsibleError("Error while fetching %s: %s" % (api_url, to_native(e)))
+
+
+class InventoryModule(BaseInventoryPlugin):
+    NAME = 'vultr'
+
+    def parse(self, inventory, loader, path, cache=True):
+        super(InventoryModule, self).parse(inventory, loader, path)
+        self._read_config_data(path=path)
+
+        api_account = self.get_option('api_account') or 'default'
+        conf = _load_conf(self.get_option('api_config'), api_account)
+
+        try:
+            api_key = self.get_option('api_key') or conf.get('key')
+        except Exception as e:
+            raise AnsibleError('Could not find an API key. Check the configuration files.')
+
+        hostname_preference = self.get_option('hostname') or 'v4_main_ip'
+
+        for server in _retrieve_servers(api_key):
+            server = Vultr.normalize_result(server, SCHEMA)
+            for group in ['region', 'os']:
+                self.inventory.add_group(group=server[group])
+                self.inventory.add_host(group=server[group], host=server['name'])
+
+            for attribute, value in server.items():
+                self.inventory.set_variable(server['name'], attribute, value)
+            self.inventory.set_variable(server['name'], 'ansible_host', server[hostname_preference])
```

Comment:
When `hostname_preference` is equal to `name`, there is no need to define `ansible_host`.